src/test/java/com/google/sps/MatchServletTest.java

```diff
@@ -133,6 +137,41 @@ public void nullParameterValues_ShouldNotThrowError() throws IOException, Servle
         verify(response, times(1)).sendRedirect("/logged_in_homepage.html");
     }
 
+    @Test
+    public void doGetRequestReturnMatches() throws IOException, ServletException {
+        when(sessionContext.isUserLoggedIn()).thenReturn(true);
+        when(sessionContext.getLoggedInUser()).thenReturn(testUser);
+
+        Collection<User> matches = matchRepository.getMatchesForUser(testUser);
+        String expected = gson.toJson(matches);
+    
+        //call doGet
+        String result = doGetHelper(request, response, matchServlet);
+
+        Assert.assertEquals(expected, result);
+    }
+    @Test
+    public void doGetRequestReturnError() throws IOException, ServletException {
+        when(sessionContext.isUserLoggedIn()).thenReturn(true);
+        when(sessionContext.getLoggedInUser()).thenReturn(testUser);
+
+        doGetHelper(request, response, matchServlet);
```

Comment:
Testing in this way, by adding 10 requests and then checking if the result of the 11th is false is a bit brittle (what if someone changes the class later to allow 20 requests instead of 10? Any time they change it they'd need to fix this test).

Adding multiple requests like this shouldn't be needed once you fix the injection issue (since then you could use the abuseFeature mock to force a failure).

Reply:
Implemented this!